includes/class-http.php

```diff
@@ -235,6 +235,17 @@ public static function generate_cache_key( $url ) {
 	 * @return array|WP_Error The Object data as array or WP_Error on failure.
 	 */
 	public static function get_remote_object( $url_or_object, $cached = true ) {
+		/**
+		 * Filters the URL before making a GET request.
+		 *
+		 * @param array|string|null $response      The response.
+		 * @param array|string|null $url_or_object The Object or the Object URL.
+		 */
+		$response = apply_filters( 'activitypub_pre_http_get_remote_object', null, $url_or_object );
+		if ( $response && ! \is_wp_error( $response ) ) {
```

Comment:
```suggestion
		if ( null !== $response ) {
```
If this is filtered, `WP_Error` should probably be as valid a response as any